ansible/roles/stack-sunbird/tasks/service_stack.yml

```diff
@@ -3,42 +3,43 @@
   template: src="stack_{{service_name}}.yml" dest="/home/deployer/stack/stack_{{service_name}}.yml" mode=0644
 
 - name: Save config file
-  copy: src="{{inventory_dir}}/../../../sunbird-portal/src/app/dist/index_cdn.ejs" dest="/home/deployer/config/index_cdn.ejs" mode=0644
+  copy: src="{{cdn_file_path}}" dest="/home/deployer/config/index_cdn.ejs" mode=0644
   when: sunbird_portal_cdn_url is defined
   register: copy_file
   ignore_errors: yes
 
-- name: Remove the cdn file from  player if it's present
-  shell: docker service update --config-rm index.ejs player_player
-  when: copy_file.changed
+- debug:
+    msg: "{{copy_file}}"
+
+- name: check if player is running
+  shell: docker service ps player_player
+  become: yes
+  register: playerRunning
   ignore_errors: yes
 
+- name: Remove the cdn file from  player if it's present
+  shell: docker service update --config-rm index_cdn.ejs player_player
+  when: copy_file.changed and playerRunning.changed and sunbird_portal_cdn_url is defined
```

Comment:
`when: copy_file.changed and sunbird_portal_cdn_url is defined and playerRunning.rc == 0`